libass/ass_render.c

```diff
@@ -29,6 +29,7 @@
 #include "ass_parse.h"
 #include "ass_priv.h"
 #include "ass_shaper.h"
+#include "ass_utils.h"
```

Comment:
I'm not a fan of excessive includes. `ass_render.c` already heavily depends on `ass_utils.h`, so adding one constant changes nothing.

Reply:
I now see `ass_render.h` already includes `ass_utils.h`, which is definitely **not** where I'd expect this to be, as nothing in the header itself requires `ass_utils.h`.
Imo, a header file should only include the bare minimum it requires for the header to be usable, so if it eg declares a function returning `size_t`, it should include `<stddef.h>`, but *not* things only the associated source file needs. Each source file should include everything it uses directly, except for types etc inherited from its own header file.
This has the advantage of making implicit dependencies sneaking in less likely. If a header file removes an include, because its own source file no longer needs it, with implicit dependencies now suddenly *other* source files might break, because they relied on that implicit/transitive include. Also putting the includes for the *source* file in the *header* file, means a lot of unneeded garbage gets included in other source files including the header, which also results in unnecessary recompiles when *any* header gets edited.
Currently it seems libass' includes are just placed wherever, or is there some system to it?

… But this pr is not about cleaning up includes, so I just removed `ass_utils.h` from `ass_render.c` for now.

Reply:
> I now see `ass_render.h` already includes `ass_utils.h`

It's also indirectly included dozen times through `ass_outline.h`.

> Currently it seems libass' includes are just placed wherever, or is there some system to it?

I've tried to minimize include count in [this PR](https://github.com/libass/libass/pull/430) but hadn't seen much enthusiasm about it.